doc/ant-xgboost_user_guide.md

```diff
@@ -0,0 +1,265 @@
+    
+    In Ant-XGBoost, we validate parameters to make sure all parameters are consistent with each other, e.g., `num_class = 3` and `objective = binary:logistic` will raise an exception. 
+    
+    In addition, we try our best to understand the input parameters from the user and automatically set or rewrite some parameters in `auto_train` mode. 
+    For example, when the feature dimension is very high, building a single tree will be very inefficient,
+    we will automatically set `colsample_bytree` and make sure at most 2000 features are used to build for each tree. Note that automatic parameter rewritting is only turned on in `auto_train` mode, in standard `train`, we only validate parameters and the behavior is fully controled by the user.
+
+* automatic training
+
+    With convergence test, number of trees in a boosted ensemble becomes a less important parameter; 
+    one can always set a very large number and rely on convergence test to figure out the right number of trees. 
+    The most important parameters to tune in boosted trees now become learning rate and max depth. 
+    In Ant-XGBoost, we employ grid search with early stopping to efficiently search the best model structure; 
+    unpromising learning rate or depth will be skipped entirely.
+    
+    While the current `auto_train` method is a very simple approach, we are working on better strategies to further scale up hyperparameter tuning in XGBoost training.
+
+
+### Helpful Backports of XGBoost master
+We keep watching the progress of XGBoost community.
+Our stable version is based on latest stable version of dmlc/xgboost.
+What's more, we will cherry-pick helpful cutting edge commits, such as blocking issue fix and non-trivial performance improvement.
+
+<br>
+
+## Quick Start
+
+* train XGBoost model on [boston housing dataset](https://www.kaggle.com/c/boston-housing)
+
+```sql
+SELECT *
+FROM boston.train
+TRAIN xgboost.Regressor
+WITH
+    train.num_round = 30,
+    train.eta = 0.15,
+    train.max_depth = 5,
+    train.subsample = 0.9
+COLUMN crim, zn, indus, chas, nox, rm, age, dis, rad, tax, ptratio, b, lstat
+LABEL medv
+INTO sqlflow_models.my_boston_xgb_model;
+```
+
+* predict with trained XGBoost model
+
+```sql
+SELECT *
+FROM boston.test
+PREDICT boston.prediction.result
+WITH
+    pred.append_columns = [crim, zn, indus, chas, nox, rm, age, dis, rad, tax, ptratio, b, lstat, medv]
+USING sqlflow_models.my_boston_xgb_model;
+```
+
+* show some prediction results
+
+```sql
+SELECT * FROM boston.prediction LIMIT 10;
+```
+
+* try auto train 
+
+```sql
+SELECT *
+FROM boston.train
+TRAIN xgboost.Regressor
+WITH
+    train.auto_train = true
+COLUMN crim, zn, indus, chas, nox, rm, age, dis, rad, tax, ptratio, b, lstat
+LABEL medv
+INTO sqlflow_models.my_boston_autoxgb_model;
+```
+
+<br>
+
+## Overall SQL Syntax
+### Training Syntax
+```sql
+// standard select clause
+SELECT ... FROM ${TABLE_NAME}
+// train clause
+TRAIN xgboost.${estimatorType}
```

Comment:
Here mentioned `estimatorType`, but the concept estimator is not yet introduced here. I see it will be introduced in the next section. Do you think it could be more comprehensive if we move the next section before this one?